bin/impl/setup.sh

```diff
@@ -36,10 +36,15 @@ verify_exist_hash $HADOOP_TARBALL $HADOOP_MD5
 verify_exist_hash $ZOOKEEPER_TARBALL $ZOOKEEPER_MD5
 verify_exist_hash $SPARK_TARBALL $SPARK_MD5
 
-host `hostname` &> /dev/null
-if [ $? != 0 ]; then
-  echo "ERROR - Your machine failed to do a DNS lookup of your IP given your hostname using 'host `hostname`'.  This is likely a DNS issue"
-  echo "that can cause fluo-dev services (such as Hadoop) to not start up.  You should confirm that /etc/resolv.conf is correct."
+hostname=$(hostname)
+if [ `grep -c "${hostname}" /etc/hosts` -ge 1 ]; then
+  echo "Found ${hostname} in /etc/hosts."
+elif [ `host "${hostname}"` -ge 1 ]; then
```

Comment:
I tested this on my machine and this line was giving my issues.  I think the return code should be checked rather than output of the `host ${hostname}` command.  I modified the script and the following code worked for me:

```bash
hostname=$(hostname)
if [ `grep -c "${hostname}" /etc/hosts` -ge 1 ]; then
  echo "Found ${hostname} in /etc/hosts."
else
  host ${hostname} &> /dev/null
  if [ $? -eq 0 ]; then
    echo "Found ${hostname} in DNS."
  else
    echo "ERROR - Your machine was unable to find its own hostname in /etc/hosts or by using 'host $hostname'."
    echo "This is an issue that can cause fluo-dev services (such as Hadoop) to not start up.  You should"
    echo "confirm that there is an entry in /etc/hosts or that /etc/resolv.conf is correct."
    exit 1
  fi
fi
```